modules/core/src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -554,6 +547,10 @@ object KafkaConsumer {
 
       override def toString: String =
         "KafkaConsumer$" + id
+
+      override def terminate: F[Unit] = fiber.cancel
```

Comment:
Perhaps this should be 
```suggestion
      override def terminate: F[Unit] = polls.cancel *> actor.cancel
```
or
```suggestion
      override def terminate: F[Unit] = polls.cancel.start.void *> actor.cancel.start.void
```

Reply:
I *think* that should be ok. Only thing is that whether `cancel` blocks is implementation-dependent - it doesn't with `IO`, but it can with other implementations of the CE typeclasses.

I think the worst that could cause would be slower shutdown rather than deadlock, but again I'm not certain.

Maybe we should start with inclining the `combine` stuff from `Fiber` and go from there.

Reply:
> Only thing is that whether cancel blocks is implementation-dependent - it doesn't with IO, but it can with other implementations of the CE typeclasses.

It does with `ce2.IO`, but not always due to implementation quirks, it always does with `ce3.IO`.  

I don't know what the rationale is here for not awaiting for shutdown, but the reason we have await-cancel  semantics in CE is that doing `cancel.start.void` at the call site to avoid waiting is pretty easy, whereas if you don't wait, and call-site wants to, you're in trouble

Reply:
>  doing cancel.start.void at the call site to avoid waiting is pretty easy, whereas if you don't wait, and call-site wants to, you're in trouble

Can't you do `fiber.cancel *> fiber.join.void` at the call site in that case?

Either way, I agree it's probably better to have `terminate` await termination - I mainly just wanted to avoid changing both API and behaviour in the same PR. It would be interesting to hear any thoughts from @vlovgr or @LMnet on this.

Reply:
> Can't you do fiber.cancel *> fiber.join.void at the call site in that case?


if the implementation of `cancel` does not wait for the resource to close (e.g. it's `thing.close.start.void`), and you want to wait for that at call site, there's nothing you can do as you've lost a handle on the fiber.

Also in ce2 if you do `fiber.cancel *> fiber.join` , it's going to be non-terminating, and in ce3 you're going to have an `Outcome`, but that still doesn't change anything in the general case unless `join` happens to wait for the resource closing (which I think it's the case here), but then just having one method to "cancel and wait", which you can `start`, seems easier

Reply:
The reason why we don't use the real `Fiber#join` and `cancel` here is due to the quirky behaviours @SystemFw describe. I'm not sure why `cancel` doesn't await termination here, but I don't see a good reason for it. I'm all for changing it. 👍

Reply:
Ok, shall we change terminate to `fiber.cancel *> awaitTermination` and merge this?